### PR TITLE
Change login flow, use public api endpoints, support on-premises instances

### PR DESCRIPTION
**Pull request by MB-Finski**
Changes the UI login flow to better match that of the Collaboard website. Also, provides a fix for the breaking change in the upstream authentication logic.

P.S. Just realized looking at the change highlights that the collaboardApiService->login() method is a bit "heavy" and could maybe benefit from some refactoring, wdyt?

### CODE REVIEW

**Comment by MB-Finski** [comment]
@julien-nc Both prs are now consolidated under this pr. Did a quick test and login and basic functionalities work now so you should be able to test the private instance.

File: lib/Controller/ConfigController.php [hunk 1]
```diff
@@ -16,26 +17,33 @@
...
 use OCP\AppFramework\Controller;
+use Psr\Log\LoggerInterface;
 
 use OCA\Collaboard\Service\CollaboardAPIService;
 use OCA\Collaboard\AppInfo\Application;
 use OCP\PreConditionNotMetException;
 
-class ConfigController extends Controller {
+class ConfigController extends Controller
+{
```
Comment (julien-nc) [inline lib/Controller/ConfigController.php]: Nitpicking: Let's stick with inline opening block character everywhere.
```suggestion
class ConfigController extends Controller {
```

**Review by julien-nc — COMMENTED** [review]
Noice!
Didn't fully review the code yet.
It works with a self hosted instance (at least their proxmox one). I had to find the base URL from the browser dev tools (again) because when the instance can be browsed at `https://whatever.org`, the API base URL is `https://whatever.org/server`. I don't know if it's systematically like that...

A few things:
* The thumbnail endpoint does not work (in both main and proxmox instances), It fails with `400 Authentication information is not given in the correct format. Check the value of Authorization header`
* We need the `acceptProjectInvitation` base URL as a setting (even if in the proxmox instance it's the same as the web base URL). Detail: in the proxmox instance, the generated links point to the main instance which does not work. Manually changing the base URL works...
* We should stop storing the login and password
    * The password is explicitly stored in CollaboardApiService::login
    * Both login and password are stored in ConfigController::setConfig. We should just return after calling the login method IMO.

File: lib/Controller/ConfigController.php (hunk 1 above)
Comment (MB-Finski) [reply lib/Controller/ConfigController.php]: Argh.. That the default formatting option for the php plugin I'm using with VSC. I'll set up composer to get cs:fix going...

**Comment by MB-Finski** [comment]
@julien-nc It should be, functionality-wise, pretty much done now and ready for a review..

File: src/components/AdminSettings.vue [hunk 2]
```diff
@@ -12,16 +12,48 @@
 				</label>
 				<input id="collaboard-instance"
 					v-model="state.admin_instance_url"
+					:class="{ 'greyed-out-text': !usingCustomInstanceUrl, 'invalid-url': !isValidUrl(state.admin_instance_url) }"
```
Comment (julien-nc) [inline src/components/AdminSettings.vue]: Would be cleaner with a computed prop like `isInstanceUrlValid` which calls `isValidUrl`. It's not clear that Vue is reactive when written like it is now.

File: src/components/AdminSettings.vue [hunk 3]
```diff
@@ -12,16 +12,48 @@
...
+				</transition>
+			</div>
+			<div class="field">
+				<label for="collaboard-invite-url">
+					<EarthIcon :size="20" class="icon" />
+					{{ t('integration_collaboard', 'Default Collaboard invite url') }}
+				</label>
+				<input id="collaboard-instance"
+					v-model="state.admin_invite_url"
+					:class="{ 'greyed-out-text': !usingCustomInviteUrl, 'invalid-url': !isValidUrl(state.admin_invite_url) }"
```
Comment (julien-nc) [inline src/components/AdminSettings.vue]: Same here, computed `isInviteUrlValid`.

**Review by julien-nc — CHANGES_REQUESTED** [review]
Minor remark. Then i'll stop nitpicking.
Super nice changes. I'll update `@nextcloud/vue` in a future PR.

File: src/components/AdminSettings.vue (hunk 2 above)
Comment (MB-Finski) [reply src/components/AdminSettings.vue]: Ah, didn't quite think that one through! But seems likely it won't/might not be responsive.

File: src/components/AdminSettings.vue (hunk 3 above)
Comment (MB-Finski) [reply src/components/AdminSettings.vue]: Fixed in https://github.com/nextcloud/integration_collaboard/pull/19/commits/0ced9cc9128977153b487efce91e544f246bc0d0

**Review by julien-nc — APPROVED** [review]
:rocket:

**Pull request merged**